### DISCUSSION TITLE
How to reduce the RMSE in energy without losing accuracy in force

### DISCUSSION THREAD

**Discussion by lnnbig** [deepmodeling/deepmd-kit#1929] [Q&A]
I trained my NNP model with the following loss parameters (see a full input.json file at the end):
"loss": {
        "start_pref_e": 0.02,
        "limit_pref_e": 1,
        "start_pref_f": 1000,
        "limit_pref_f": 1,
        "start_pref_v": 10,
        "limit_pref_v": 0.2
}.
The final energy and force error (RMSE) is 10-2 eV/atom and 10-1 eV/angstrom, as shown in the attached figure. I want to control the energy error below 10-3 eV/atom. Perhaps I can increase "limit_pref_e", but this means that I will sacrifice the accuracy in force.
 
Do you have any suggestions to reduce the RMSE in energy without losing accuracy in force? 
Thanks very much.

![RMSE](https://user-images.githubusercontent.com/105722950/190970312-84f2eb62-bd37-4612-b3d6-6af86e7f2188.png)

### input.json ###

{
    "model": {
        "descriptor": {
            "type": "se_e2_a",
            "sel": "auto",
            "rcut_smth": 0.5,
            "rcut": 6.5,
            "neuron": [
                25,
                50,
                100
            ],
            "axis_neuron": 12,
            "seed": 3482162051
        },
        "fitting_net": {
            "neuron": [
                240,
                240,
                240
            ],
            "resnet_dt": true,
            "seed": 3277049792
        },
        "type_map": [
            "XXXX",
            "XXXX"
        ]
    },
    "learning_rate": {
        "type": "exp",
        "decay_steps": 5000,
        "start_lr": 0.001,
        "stop_lr": 3.51e-08
    },
    "loss": {
        "start_pref_e": 0.02,
        "limit_pref_e": 1,
        "start_pref_f": 1000,
        "limit_pref_f": 1,
        "start_pref_v": 10,
        "limit_pref_v": 0.2
    },
    "training": {
        "numb_steps": 300000,
        "disp_file": "lcurve.out",
        "disp_freq": 1000,
        "save_freq": 10000,
        "disp_training": true,
        "time_training": true,
        "profiling": false,
        "profiling_file": "timeline.json",
        "seed": 1799803115,
        "training_data": {
            "systems": [
                "../data.init/set21",
                "../data.init/set25",
                "../data.init/set26",
                "../data.init/set27",
                "../data.iters/iter.000000/02.fp/data.058",
                "../data.iters/iter.000000/02.fp/data.047",
                "../data.iters/iter.000000/02.fp/data.008",
                "../data.iters/iter.000000/02.fp/data.056",
                "../data.iters/iter.000000/02.fp/data.059",
                "../data.iters/iter.000000/02.fp/data.051",
                "../data.iters/iter.000000/02.fp/data.020",
                "../data.iters/iter.000000/02.fp/data.007",
                "../data.iters/iter.000000/02.fp/data.060",
                "../data.iters/iter.000000/02.fp/data.013",
                "../data.iters/iter.000000/02.fp/data.042",
                "../data.iters/iter.000000/02.fp/data.052",
                "../data.iters/iter.000000/02.fp/data.027",
                "../data.iters/iter.000000/02.fp/data.026",
                "../data.iters/iter.000000/02.fp/data.033",
                "../data.iters/iter.000000/02.fp/data.016",
                "../data.iters/iter.000000/02.fp/data.044",
                "../data.iters/iter.000000/02.fp/data.038",
                "../data.iters/iter.000000/02.fp/data.022",
                "../data.iters/iter.000000/02.fp/data.006",
                "../data.iters/iter.000000/02.fp/data.024",
                "../data.iters/iter.000000/02.fp/data.018",
                "../data.iters/iter.000000/02.fp/data.040",
                "../data.iters/iter.000000/02.fp/data.050",
                "../data.iters/iter.000000/02.fp/data.010",
                "../data.iters/iter.000000/02.fp/data.028",
                "../data.iters/iter.000000/02.fp/data.045",
                "../data.iters/iter.000000/02.fp/data.053",
                "../data.iters/iter.000000/02.fp/data.061",
                "../data.iters/iter.000000/02.fp/data.034",
                "../data.iters/iter.000000/02.fp/data.048",
                "../data.iters/iter.000000/02.fp/data.036",
                "../data.iters/iter.000000/02.fp/data.049",
                "../data.iters/iter.000000/02.fp/data.054",
                "../data.iters/iter.000000/02.fp/data.025",
                "../data.iters/iter.000000/02.fp/data.043",
                "../data.iters/iter.000000/02.fp/data.057",
                "../data.iters/iter.000000/02.fp/data.001",
                "../data.iters/iter.000000/02.fp/data.062",
                "../data.iters/iter.000000/02.fp/data.037",
                "../data.iters/iter.000000/02.fp/data.063",
                "../data.iters/iter.000000/02.fp/data.030",
                "../data.iters/iter.000000/02.fp/data.041",
                "../data.iters/iter.000000/02.fp/data.004",
                "../data.iters/iter.000000/02.fp/data.011",
                "../data.iters/iter.000000/02.fp/data.003",
                "../data.iters/iter.000000/02.fp/data.017",
                "../data.iters/iter.000000/02.fp/data.023",
                "../data.iters/iter.000000/02.fp/data.039",
                "../data.iters/iter.000000/02.fp/data.031",
                "../data.iters/iter.000000/02.fp/data.046",
                "../data.iters/iter.000000/02.fp/data.055",
                "../data.iters/iter.000000/02.fp/data.029",
                "../data.iters/iter.000000/02.fp/data.000",
                "../data.iters/iter.000001/02.fp/data.058",
                "../data.iters/iter.000001/02.fp/data.047",
                "../data.iters/iter.000001/02.fp/data.056",
                "../data.iters/iter.000001/02.fp/data.059",
                "../data.iters/iter.000001/02.fp/data.015",
                "../data.iters/iter.000001/02.fp/data.005",
                "../data.iters/iter.000001/02.fp/data.051",
                "../data.iters/iter.000001/02.fp/data.020",
                "../data.iters/iter.000001/02.fp/data.007",
                "../data.iters/iter.000001/02.fp/data.060",
                "../data.iters/iter.000001/02.fp/data.013",
                "../data.iters/iter.000001/02.fp/data.042",
                "../data.iters/iter.000001/02.fp/data.052",
                "../data.iters/iter.000001/02.fp/data.027",
                "../data.iters/iter.000001/02.fp/data.009",
                "../data.iters/iter.000001/02.fp/data.026",
                "../data.iters/iter.000001/02.fp/data.033",
                "../data.iters/iter.000001/02.fp/data.016",
                "../data.iters/iter.000001/02.fp/data.044",
                "../data.iters/iter.000001/02.fp/data.014",
                "../data.iters/iter.000001/02.fp/data.038",
                "../data.iters/iter.000001/02.fp/data.022",
                "../data.iters/iter.000001/02.fp/data.006",
                "../data.iters/iter.000001/02.fp/data.024",
                "../data.iters/iter.000001/02.fp/data.018",
                "../data.iters/iter.000001/02.fp/data.021",
                "../data.iters/iter.000001/02.fp/data.040",
                "../data.iters/iter.000001/02.fp/data.050",
                "../data.iters/iter.000001/02.fp/data.010",
                "../data.iters/iter.000001/02.fp/data.019",
                "../data.iters/iter.000001/02.fp/data.028",
                "../data.iters/iter.000001/02.fp/data.045",
                "../data.iters/iter.000001/02.fp/data.053",
                "../data.iters/iter.000001/02.fp/data.061",
                "../data.iters/iter.000001/02.fp/data.034",
                "../data.iters/iter.000001/02.fp/data.048",
                "../data.iters/iter.000001/02.fp/data.036",
                "../data.iters/iter.000001/02.fp/data.049",
                "../data.iters/iter.000001/02.fp/data.054",
                "../data.iters/iter.000001/02.fp/data.025",
                "../data.iters/iter.000001/02.fp/data.002",
                "../data.iters/iter.000001/02.fp/data.043",
                "../data.iters/iter.000001/02.fp/data.057",
                "../data.iters/iter.000001/02.fp/data.012",
                "../data.iters/iter.000001/02.fp/data.001",
                "../data.iters/iter.000001/02.fp/data.062",
                "../data.iters/iter.000001/02.fp/data.037",
                "../data.iters/iter.000001/02.fp/data.063",
                "../data.iters/iter.000001/02.fp/data.030",
                "../data.iters/iter.000001/02.fp/data.041",
                "../data.iters/iter.000001/02.fp/data.004",
                "../data.iters/iter.000001/02.fp/data.011",
                "../data.iters/iter.000001/02.fp/data.003",
                "../data.iters/iter.000001/02.fp/data.017",
                "../data.iters/iter.000001/02.fp/data.023",
                "../data.iters/iter.000001/02.fp/data.039",
                "../data.iters/iter.000001/02.fp/data.031",
                "../data.iters/iter.000001/02.fp/data.046",
                "../data.iters/iter.000001/02.fp/data.055",
                "../data.iters/iter.000001/02.fp/data.032",
                "../data.iters/iter.000001/02.fp/data.035",
                "../data.iters/iter.000001/02.fp/data.029",
                "../data.iters/iter.000001/02.fp/data.000",
                "../data.iters/iter.000002/02.fp/data.058",
                "../data.iters/iter.000002/02.fp/data.047",
                "../data.iters/iter.000002/02.fp/data.008",
                "../data.iters/iter.000002/02.fp/data.056",
                "../data.iters/iter.000002/02.fp/data.059",
                "../data.iters/iter.000002/02.fp/data.015",
                "../data.iters/iter.000002/02.fp/data.005",
                "../data.iters/iter.000002/02.fp/data.051",
                "../data.iters/iter.000002/02.fp/data.020",
                "../data.iters/iter.000002/02.fp/data.007",
                "../data.iters/iter.000002/02.fp/data.060",
                "../data.iters/iter.000002/02.fp/data.013",
                "../data.iters/iter.000002/02.fp/data.042",
                "../data.iters/iter.000002/02.fp/data.052",
                "../data.iters/iter.000002/02.fp/data.027",
                "../data.iters/iter.000002/02.fp/data.009",
                "../data.iters/iter.000002/02.fp/data.026",
                "../data.iters/iter.000002/02.fp/data.033",
                "../data.iters/iter.000002/02.fp/data.016",
                "../data.iters/iter.000002/02.fp/data.044",
                "../data.iters/iter.000002/02.fp/data.014",
                "../data.iters/iter.000002/02.fp/data.038",
                "../data.iters/iter.000002/02.fp/data.022",
                "../data.iters/iter.000002/02.fp/data.006",
                "../data.iters/iter.000002/02.fp/data.024",
                "../data.iters/iter.000002/02.fp/data.021",
                "../data.iters/iter.000002/02.fp/data.040",
                "../data.iters/iter.000002/02.fp/data.050",
                "../data.iters/iter.000002/02.fp/data.010",
                "../data.iters/iter.000002/02.fp/data.019",
                "../data.iters/iter.000002/02.fp/data.028",
                "../data.iters/iter.000002/02.fp/data.045",
                "../data.iters/iter.000002/02.fp/data.053",
                "../data.iters/iter.000002/02.fp/data.061",
                "../data.iters/iter.000002/02.fp/data.034",
                "../data.iters/iter.000002/02.fp/data.048",
                "../data.iters/iter.000002/02.fp/data.036",
                "../data.iters/iter.000002/02.fp/data.049",
                "../data.iters/iter.000002/02.fp/data.054",
                "../data.iters/iter.000002/02.fp/data.025",
                "../data.iters/iter.000002/02.fp/data.043",
                "../data.iters/iter.000002/02.fp/data.057",
                "../data.iters/iter.000002/02.fp/data.012",
                "../data.iters/iter.000002/02.fp/data.062",
                "../data.iters/iter.000002/02.fp/data.037",
                "../data.iters/iter.000002/02.fp/data.063",
                "../data.iters/iter.000002/02.fp/data.030",
                "../data.iters/iter.000002/02.fp/data.041",
                "../data.iters/iter.000002/02.fp/data.004",
                "../data.iters/iter.000002/02.fp/data.011",
                "../data.iters/iter.000002/02.fp/data.023",
                "../data.iters/iter.000002/02.fp/data.039",
                "../data.iters/iter.000002/02.fp/data.031",
                "../data.iters/iter.000002/02.fp/data.046",
                "../data.iters/iter.000002/02.fp/data.055",
                "../data.iters/iter.000002/02.fp/data.032",
                "../data.iters/iter.000002/02.fp/data.035",
                "../data.iters/iter.000002/02.fp/data.029",
                "../data.iters/iter.000002/02.fp/data.000",
                "../data.iters/iter.000003/02.fp/data.008",
                "../data.iters/iter.000003/02.fp/data.015",
                "../data.iters/iter.000003/02.fp/data.005",
                "../data.iters/iter.000003/02.fp/data.020",
                "../data.iters/iter.000003/02.fp/data.007",
                "../data.iters/iter.000003/02.fp/data.013",
                "../data.iters/iter.000003/02.fp/data.027",
                "../data.iters/iter.000003/02.fp/data.009",
                "../data.iters/iter.000003/02.fp/data.026",
                "../data.iters/iter.000003/02.fp/data.016",
                "../data.iters/iter.000003/02.fp/data.014",
                "../data.iters/iter.000003/02.fp/data.022",
                "../data.iters/iter.000003/02.fp/data.006",
                "../data.iters/iter.000003/02.fp/data.024",
                "../data.iters/iter.000003/02.fp/data.018",
                "../data.iters/iter.000003/02.fp/data.021",
                "../data.iters/iter.000003/02.fp/data.010",
                "../data.iters/iter.000003/02.fp/data.019",
                "../data.iters/iter.000003/02.fp/data.028",
                "../data.iters/iter.000003/02.fp/data.025",
                "../data.iters/iter.000003/02.fp/data.002",
                "../data.iters/iter.000003/02.fp/data.012",
                "../data.iters/iter.000003/02.fp/data.001",
                "../data.iters/iter.000003/02.fp/data.030",
                "../data.iters/iter.000003/02.fp/data.004",
                "../data.iters/iter.000003/02.fp/data.011",
                "../data.iters/iter.000003/02.fp/data.003",
                "../data.iters/iter.000003/02.fp/data.017",
                "../data.iters/iter.000003/02.fp/data.023",
                "../data.iters/iter.000003/02.fp/data.031",
                "../data.iters/iter.000003/02.fp/data.029",
                "../data.iters/iter.000003/02.fp/data.000"
            ],
            "batch_size": [
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1
            ]
        }
    }
}

**Comment by wanghan-iapcm** [comment]
Have you tried to use a larger E prefactor? How much force accuracy would you loss?
